### PR TITLE
xen-guest-agent: 0.3.0 -> 0.4.0-unstable-2024-05-31

### PR DESCRIPTION
**Pull request by SigmaSquadron**
## Description of changes

- Moves package to `by-name`.
- Formats `xen-guest-agent/default.nix` with `nixfmt-rfc-style`.
- Updates `xen-guest-agent` to [0.4.0](https://gitlab.com/xen-project/xen-guest-agent/-/releases/0.4.0) + commit [`03aaadbe`](https://gitlab.com/xen-project/xen-guest-agent/-/commit/03aaadbe030f303b1503e172ee2abb6d0cab7ac6) which includes a crate update to fix compilation on Rust 1.80.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  - No automated tests, as `xen-guest-agent` must be ran inside a Xen domU.
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 332306"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Change is minor, so no changelog.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SigmaSquadron** [comment]
Result of `nixpkgs-review pr 332306` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xen-guest-agent</li>
  </ul>
</details>

**Review by matdibu — APPROVED** [review]
thank you!

**Comment by SigmaSquadron** [comment]
Soooo. Rust 1.80. Right.

@matdibu: Have you [decided](https://github.com/NixOS/nixpkgs/issues/332957#issuecomment-2288336452) what's the way to go here? I have an update to `0.4.0-unstable-2024-05-31` ready.

File: pkgs/by-name/xe/xen-guest-agent/package.nix [hunk 1]
```diff
@@ -3,29 +3,32 @@
...
-    hash = "sha256-Csio24ofj+p0j/R0av/28P/KCNXhmcF+r8xGJEfoHjQ=";
+    hash = "sha256-kJSXF/iXH0VU3sp7yOfGw6kuQPNuueQVqIk0u/OrZqk=";
   };
 
-  cargoHash = "sha256-XWDDzSu88zCIwMuvkFjCb98DzXHvW2IP9u3EbpAMIgw=";
+  cargoHash = "sha256-JsL9/kDsVFx/AC0FXbBLvk9r7Q+daL8lvqli1fE0OE8=";
 
   env = {
     LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+    PKG_CONFIG_PATH = "${xen-slim.dev}/lib/pkgconfig";
```
Comment (alyssais) [inline pkgs/by-name/xe/xen-guest-agent/package.nix:24]: Putting it xen-slim in buildInputs isn't enough?

**Comment by matdibu** [comment]
I think it's safe to pull an unstable version with the fix already included, the risk is low enough.

File: pkgs/by-name/xe/xen-guest-agent/package.nix (hunk 1 above)
Comment (SigmaSquadron) [reply pkgs/by-name/xe/xen-guest-agent/package.nix]: Turns out the majority of the environment variables weren't needed. Thanks!

**Comment by SigmaSquadron** [comment]
Also tested on both a XCP-ng and a NixOS host. In both instances, the guest created the XenStore entries.

---

Result of `nixpkgs-review pr 332306` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xen-guest-agent</li>
  </ul>
</details>

**Review by matdibu — APPROVED** [review]

File: pkgs/by-name/xe/xen-guest-agent/package.nix [hunk 2]
```diff
@@ -3,32 +3,31 @@
...
 
   nativeBuildInputs = [
     llvmPackages.clang
-    xen-slim.out
+    pkg-config
   ];
 
+  buildInputs = [ xen-slim ];
+
+  env.LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
```
Comment (alyssais) [inline pkgs/by-name/xe/xen-guest-agent/package.nix:29]: Is this for bindgen?  If so, we have rustPlatform.bindgenHook.  (But it was already not in use, so doesn't have to be in scope for this PR.)

Comment (SigmaSquadron) [reply pkgs/by-name/xe/xen-guest-agent/package.nix:29]: I'll be honest; I'm not entirely sure of the specifics of the `rustPlatform` conventions. I'll read up on the available documentation and open a PR if we need to remove anything else.

**Pull request merged**